gapic/src/main/com/google/gapid/perfetto/models/PowerSummaryTrack.java

```diff
@@ -0,0 +1,184 @@
+import static com.google.gapid.perfetto.views.TrackContainer.group;
+import static com.google.gapid.perfetto.views.TrackContainer.single;
+import static com.google.gapid.util.MoreFutures.transform;
+import static com.google.gapid.util.MoreFutures.transformAsync;
+import static java.lang.String.format;
+
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gapid.models.Perfetto;
+import com.google.gapid.perfetto.Unit;
+import com.google.gapid.perfetto.models.TrackConfig.Group;
+import com.google.gapid.perfetto.views.CounterPanel;
+import com.google.gapid.perfetto.views.PowerSummaryPanel;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** {@link Track} summarizing the total Power usage. */
+public class PowerSummaryTrack extends Track.WithQueryEngine<PowerSummaryTrack.Data> {
+  private static String viewSql;
+  private static int numPowerRailTracks;
+  public static Unit unit;
+  public static Double minValue;
+  public static Double maxValue;
+
+  private static final String VIEW_SQL_MONOTONIC =
+      "select ts + 1 ts, lead(ts) over win - ts dur, lead(value) over win - value value, lead(id)"
+          + " over win id from counter where track_id = ";
+
+  public PowerSummaryTrack(QueryEngine qe, int numTracks) {
+    super(qe, "power_sum");
+    this.numPowerRailTracks = numTracks;
+  }
+
+  public int getNumPowerRailTracks() {
+    return numPowerRailTracks;
+  }
+
+  public static Perfetto.Data.Builder enumerate(Perfetto.Data.Builder data) {
+    ImmutableListMultimap<String, CounterInfo> counters = data.getCounters(CounterInfo.Type.Global);
+    List<CounterInfo> powerRails =
+        counters.entries().stream()
+            .filter(entry -> entry.getKey().startsWith("power.rails"))
+            .map(Map.Entry::getValue)
+            .collect(Collectors.toList());
+
+    if (powerRails.size() == 0) {
+      return data;
+    }
+
+    // Power Group
+    viewSql = buildViewSql(powerRails);
+    PowerSummaryTrack powerSummaryTrack = new PowerSummaryTrack(data.qe, powerRails.size());
+
+    // Power Rails tracks.
+    for (CounterInfo powerRail : powerRails) {
+      unit = powerRail.unit;
+      CounterTrack powerRailTrack = new CounterTrack(data.qe, powerRail);
+      data.tracks.addTrack(
+          powerSummaryTrack.getId(),
+          powerRailTrack.getId(),
+          powerRail.name,
+          single(
+              state -> new CounterPanel(state, powerRailTrack, POWER_RAIL_COUNTER_TRACK_HEIGHT),
+              true,
+              true));
+    }
+
+    Group.UiFactory ui = group(state -> new PowerSummaryPanel(state, powerSummaryTrack), false);
+    data.tracks.addLabelGroup(null, powerSummaryTrack.getId(), "Power Usage", ui);
+    return data;
+  }
+
+  private static String buildViewSql(List<CounterInfo> powerRails) {
+    minValue = powerRails.get(0).min;
```

Comment:
Maybe, instead get the `minValue` and `maxValue` in the loop in line 80 instead. It seems to be iterating over the same data, but it's a little confusing to me for this function named `buildViewSql` to both build a sql query and also get the proper min/max value

Reply:
Yep. I should have moved it to Line 80. Thanks for pointing that out.